### ISSUE TITLE
Different results compared to the paper

### ISSUE THREAD

**Issue by prosho-97** [flipz357/bamboo-amr-benchmark#1]
Hi! Thank you for sharing your work. I tried to replicate the correlation results of the metrics tested in the TACL paper you published and only those of the `SEMA` metric are precisely the same. Those of all the other metrics change (but the final ranking of the metrics remains the same):

[bamboo_metrics.pdf](https://github.com/flipz357/bamboo-amr-benchmark/files/10863993/bamboo_metrics.pdf)

As scripts to run the metrics, I used those in this [repo](https://github.com/Heidelberg-NLP/AMRParseEval). For the Weisfeiler-Leman Graph Kernels metrics I used the conda env suggested in the [corresponding repo](https://github.com/flipz357/weisfeiler-leman-amr-metrics). For the other metrics instead, this is the conda env I used:
```
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - ca-certificates=2023.01.10=h06a4308_0
  - certifi=2022.12.7=py39h06a4308_0
  - ld_impl_linux-64=2.38=h1181459_1
  - libffi=3.4.2=h6a678d5_6
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libstdcxx-ng=11.2.0=h1234567_1
  - ncurses=6.4=h6a678d5_0
  - openssl=1.1.1t=h7f8727e_0
  - pip=22.3.1=py39h06a4308_0
  - python=3.9.16=h7a1cb2a_0
  - readline=8.2=h5eee18b_0
  - setuptools=65.6.3=py39h06a4308_0
  - sqlite=3.40.1=h5082296_0
  - tk=8.6.12=h1ccaba5_0
  - tzdata=2022g=h04d1e81_0
  - wheel=0.37.1=pyhd3eb1b0_0
  - xz=5.2.10=h5eee18b_1
  - zlib=1.2.13=h5eee18b_0
  - pip:
      - numpy==1.24.2
      - scipy==1.10.0
```

Afaik, `simpleMetric`, `SEMA`, `SEMBLEU`, and `WLK_k2` should be the only deterministic metrics among those in the file above. In fact, `SEMA` gets exactly the same correlations, and `SEMBLEU` has small variations (I didn't compute the symmetric version), but I don't know why with `WLK_k2` I got different correlations (without taking into account correlations related to argument confusion, as the way of computing them has changed).

**Comment by flipz357** [comment]
Hi Lorenzo,

thanks for your input, interesting observations!

Re Sembleu:

- i): Maybe I had computed it in the other direction? This could explain some small variations.
- ii): It is interesting that you observe minor variations across runs, there shouldn't be randomness as far as I know. Can you replicate different predictions for a single pair?

Re S(2)match:

- I think the small deviations can be due to the hill-climbing. 

Re WLK:

- It seems it is now a bit better on STS and SICK and a bit worse on PARA. Maybe there was some change in the code. From the original code to the code that you used there were a few re-factorings/cleanings.

Re WWLK:

- i) See WLK.
- ii) The randomness in WWLK is due to node/edge embedding initialization for which we cannot find something in the pre-trained embeddings (UNK). To make it more deterministic I added the option to calculate expected distance (matrix) over different random seeds (maybe this makes it also a bit better). To make it 100% deterministic I think one would have to manually fix parameters for UNK (which seems kind of unappealing) or use an embedding base that returns an embedding for every input. Maybe there are other ways.

[Re ARG test: If a and b are paraphrases we construct -a by changing meaning of a. So we want score(a,b) > score(-a, b). Accuracy seems like a somewhat better way to measure this than correlation against 1 and 0. But you can set it back to correlation if you like.]

**Comment by prosho-97** [comment]
Thank you for your quick reply Juri!

For SEMBLEU I reversed the order of the files in the pairs and got exactly the same results in the paper. So now, apart from WLK (for which the difference is due to the change in the code I guess at this point), all the deterministic metrics obtained exactly the same correlations as in the paper. I was not getting variations across runs, the problem was simply that some deterministic metrics obtained different correlations wrt the paper, but now (apart from WLK), I no longer have these differences, thanks!
